java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/utils/UrlParser.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.driver.jdbc.utils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * URL Parser for extracting key values from a connection string.
+ */
+public class UrlParser {
+  /**
+   * Parse a url key value parameters.
+   *
+   * @param url {@link String}
+   * @return {@link Map}
+   */
+  public static Map<String, String> parse(String url) {
+    Map<String, String> resultMap = new HashMap<>();
+    // TODO: add support to others separators, such as ';'
+    String[] keyValues = url.split("&");
```

Comment:
It is a simple change to just make the delimiter a parameter so let's go ahead and do so.

Reply:
done